packages/app-pages/components/AppPages/index.jsx

```diff
@@ -1,45 +1,65 @@
 import React, { useEffect } from 'react';
-import { Route, Switch, Redirect } from 'react-router-dom';
+import { Redirect, Route, Switch } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import {
-  preferencesPage,
-  profilePages,
-  plansPage,
+  campaignsPage,
   newBusinessTrialists,
   newConnection,
-  campaignsPage,
+  plansPage,
+  preferencesPage,
+  profilePages,
+  profileTabPages,
   missingAccessPage,
 } from '@bufferapp/publish-routes';
+import { filterProfilesByOrg } from '@bufferapp/publish-profile-sidebar/utils';
 import PagesWithSidebar from '@bufferapp/publish-app-pages/components/PagesWithSidebar';
 import ProfilePages from '@bufferapp/publish-app-pages/components/ProfilePages';
 import Preferences from '@bufferapp/publish-preferences';
 import Plans from '@bufferapp/publish-plans';
 import DefaultPage from '@bufferapp/default-page';
 import OnboardingManager from '@bufferapp/publish-onboarding';
+import { useOrgSwitcher, useUser } from '@bufferapp/app-shell';
 // import MissingAccessPage from '../../../missing-access-page/index';
 
 const AppPages = ({
-  profiles,
+  unfilteredProfiles,
   showBusinessTrialistsOnboarding,
   profileRouteLoaded,
-  needsToSetCurrentOrg,
-  setCurrentOrganization,
-  currentOrgId,
+  orgIdFromRoute,
 }) => {
-  const hasProfiles = profiles && profiles.length > 0;
+  // Get current selected org from appshell
+  const user = useUser();
+  const selectedOrgInAppShell = user?.currentOrganization?.id;
+
+  const currentOrgId = orgIdFromRoute || selectedOrgInAppShell;
+  const needsToSelectNewOrgInAppShell =
+    selectedOrgInAppShell !== orgIdFromRoute && !!orgIdFromRoute;
```

Comment:
This line confused me a bit at first. Can we make it more clear?

```suggestion
    (selectedOrgInAppShell !== orgIdFromRoute) && (orgIdFromRoute !== undefined);
```

Adding some parens for clarity, and then I'd prefer a `!== undefined` or `!== null` check at the end (Do you know which it would be?) since the `!!` is handy but not always clear at first glance.

Reply:
Hey @hamstu, I understand the concern about clear code, but I'm not sure I see a problem with double negation. It's very useful as it considers both `undefined`, `null`, `false` and empty strings, which is a great validation here. Doing a quick search we use the double negation to get a boolean 31 times in our codebase. How would you feel if instead of changing it I added a comment explaining it? As for the parens, our linter/prettier will remove then 😅

Reply:
Thanks @anafilipadealmeida! A comment sounds like a great solution here! Maybe it was just my early-morning brain struggling to understand things 🙈